src/scanpy/preprocessing/_simple.py

```diff
@@ -722,13 +737,13 @@ def regress_out(
                 "we regress on the mean for each category."
             )
         logg.debug("... regressing on per-gene means within categories")
-        regressors = np.zeros(X.shape, dtype="float32")
+        # Create numpy array's from categorical variable
+        cats = np.int64(len(adata.obs[keys[0]].cat.categories))
```

Comment:
Ditto

Reply:
Also comment why `np.int64`

Reply:
because it has be done because of weird typing from pandas. So this ensures that it works within the kernel

Reply:
so `len` doesn’t return a Python `int`? That’s a pandas bug.